### ISSUE TITLE
Style/YodaCode false positive on `Regexp#===`

### ISSUE THREAD

**Issue by kddeisz** [bbatsov/rubocop#4726]
## Expected behavior

I frequently use `/a/ === 'a'` as opposed to `/a/.match?('a')` because the former returns `true` or `false` while the latter returns either `MatchData` or `nil`. This should not trigger the `YodaCode` cop.

## Actual behavior

It does trigger the cop, and then says to reverse the operands. This is bad advice as `Regexp#===` and `String#===` are very different methods. Demonstrated below:

```
irb(main):001:0> 'a' === /a/
=> false
irb(main):002:0> /a/ === 'a'
=> true
```

## Steps to reproduce the problem

- Put `/a/ === 'a'` into a source file

## RuboCop version

```
$ rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by koic** [comment]
I think that this issue was resolved by #4688. Let's wait for a new release of RuboCop.

**Comment by kddeisz** [comment]
Yup sorry didn't see that.
